worlds/stardew_valley/mods/logic/item_logic.py

```diff
@@ -128,16 +128,13 @@ def get_distant_lands_item_rules(self):
             DistantLandsForageable.swamp_herb: self.logic.region.can_reach(Region.witch_swamp),
             DistantLandsForageable.brown_amanita: self.logic.region.can_reach(Region.witch_swamp),
             DistantLandsFish.purple_algae: self.logic.fishing.can_fish_at(Region.witch_swamp),
-            DistantLandsSeed.vile_ancient_fruit: self.logic.money.can_spend_at(Region.oasis, 50) & self.pseudo_cropsanity_check(DistantLandsSeed.vile_ancient_fruit),
-            DistantLandsSeed.void_mint: self.logic.money.can_spend_at(Region.oasis, 80) & self.pseudo_cropsanity_check(DistantLandsSeed.void_mint),
-            DistantLandsCrop.void_mint: self.logic.has(DistantLandsSeed.void_mint),
-            DistantLandsCrop.vile_ancient_fruit: self.logic.has(DistantLandsSeed.vile_ancient_fruit)
+            DistantLandsSeed.vile_ancient_fruit: self.logic.money.can_spend_at(Region.oasis, 50) & self.seed_check_if_usual_source(DistantLandsSeed.vile_ancient_fruit),
+            DistantLandsSeed.void_mint: self.logic.money.can_spend_at(Region.oasis, 80) & self.seed_check_if_usual_source(DistantLandsSeed.void_mint),
+            DistantLandsCrop.void_mint: self.logic.has(DistantLandsSeed.void_mint) & self.logic.season.has_any_not_winter(),
```

Comment:
Season checks should be placed before the "Has()" checks, due to it being a faster short-circuit.